platform/platform-tests/testSrc/com/intellij/openapi/editor/impl/AccessibleTextTest.java

```diff
@@ -0,0 +1,164 @@
+// Copyright 2000-2023 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license.
+package com.intellij.openapi.editor.impl;
+
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.actions.CaretStopOptions;
+import com.intellij.openapi.editor.actions.CaretStopOptionsTransposed;
+import com.intellij.openapi.editor.ex.EditorSettingsExternalizable;
+
+import javax.accessibility.AccessibleText;
+
+public class AccessibleTextTest extends AbstractEditorTest {
+  private static final String text = """
+      public static void printWelcomeMessage() {
+        System.out.println("Hello, World!");
+      }
+    """;
+
+
+  public void testGetAfterIndexReturnsCorrectWord() {
+    int firstWordOffset = 3;
+    var stopOptions = CaretStopOptionsTransposed.DEFAULT_WINDOWS.toCaretStopOptions();
+    var editor = initEditor(stopOptions, false);
+    editor.getCaretModel().moveToOffset(firstWordOffset);
+    var accessibleText = editor.getContentComponent().getAccessibleContext().getAccessibleText();
+    var expectedWord = "static";
+    var actualWord = accessibleText.getAfterIndex(AccessibleText.WORD, firstWordOffset);
+    assertEquals(expectedWord, actualWord);
+  }
+
+  public void testGetBeforeIndexReturnsCorrectWord() {
+    int secondWordOffset = 9;
+    var stopOptions = CaretStopOptionsTransposed.DEFAULT.toCaretStopOptions();
+    var editor = initEditor(stopOptions, false);
+    editor.getCaretModel().moveToOffset(secondWordOffset);
+    var accessibleText = editor.getContentComponent().getAccessibleContext().getAccessibleText();
+    var expectedWord = "public";
+    var actualWord = accessibleText.getBeforeIndex(AccessibleText.WORD, secondWordOffset);
+    assertEquals(expectedWord, actualWord);
+  }
+
+
+  public void testGetTextAtIndexReturnsCorrectWordWithDefaultStopOptionsAndCamelCaseWhenMovingForWard() {
```

Comment:
Nitpick: please change `ForWard` to `Forward` in test method names.
```suggestion
  public void testGetTextAtIndexReturnsCorrectWordWithDefaultStopOptionsAndCamelCaseWhenMovingForward() {
```